onebusaway-android/src/main/java/org/onebusaway/android/io/request/reminders/ObaReminderRequest.java

```diff
@@ -0,0 +1,122 @@
+package org.onebusaway.android.io.request.reminders;
+
+import org.onebusaway.android.R;
+import org.onebusaway.android.app.Application;
+import org.onebusaway.android.io.request.RequestBase;
+import org.onebusaway.android.io.request.reminders.model.ReminderResponse;
+
+import android.content.Context;
+import android.net.Uri;
+import android.util.Log;
+
+import java.util.concurrent.Callable;
+
+import androidx.annotation.NonNull;
+
+/**
+ * This class represent the POST request to create a arrivals alarm.
+ */
+
+public final class ObaReminderRequest extends RequestBase implements Callable<ReminderResponse> {
+
+    private final ReminderRequestListener listener;
+    private static final String TAG = "ObaReminderRequest";
+
+    private ObaReminderRequest(Uri uri, String body, ReminderRequestListener listener) {
+        super(uri, body);
+        this.listener = listener;
+    }
+
+    public static class Builder extends PostBuilderBase {
```

Comment:
I think you should add some validation of the values that are being included in the POST body, specifically verifying that all necessary values are included before the POST is sent. I'll leave it to you to determine the best location :)

Reply:
Correct.